File: c/SRC/vector/vector.c
# include "vector.h"

double	get_rad(double angle)
{
	return (angle * (PI / 180));
}

double	vec_len(double vec[2])
{
	return (sqrt(vec[X] * vec[X] + vec[Y] * vec[Y]));
}

double	vec_cos(double v1[2], double v2[2])
{
	return ((v1[X] * v2[X] + v1[Y] * v2[Y]) / (vec_len(v1) * vec_len(v2)));
}

void	vec_rot(double vec[2], double angle)
{
	double temp[2];
	
	temp[X] = vec[X] * cos(get_rad(angle))
				- vec[Y] * sin(get_rad(angle));
	temp[Y] = vec[X] * sin(get_rad(angle))
				+ vec[Y] * cos(get_rad(angle));
	vec[X] = temp[X];
	vec[Y] = temp[Y];
}

File: js/SRC/vector/vector.js
const X = 0;
const Y = 1;

exports.vecLen = (vec) => {
	return Math.sqrt(vec[X] * vec[X] + vec[Y] * vec[Y]);
}

exports.vecCos = (v1, v2) => {
	return (v1[X] * v2[X] + v1[Y] * v2[Y]) /
		(Math.sqrt(v1[X] * v1[X] + v1[Y] * v1[Y])
			* Math.sqrt(v2[X] * v2[X] + v2[Y] * v2[Y]));
}

exports.vecRotate = (vec, angle) => {
	const rad = angle * Math.PI / 180;
	const vecRot = [
		vec[X] * Math.cos(rad) - vec[Y] * Math.sin(rad),
		vec[X] * Math.sin(rad) + vec[Y] * Math.cos(rad)
	]
	return (vecRot);
}